src/utils.ts

```diff
@@ -3,7 +3,7 @@ import { IConstruct } from 'constructs';
 export function undefinedIfEmpty<T>(obj: T): T | undefined {
   if (typeof(obj) === 'string' && obj === '') { return undefined; }
   if (Array.isArray(obj) && obj.length === 0) { return undefined; }
-  if (typeof(obj) === 'object' && (Object.keys(obj).length === 0 || Object.values(obj).filter(x => x).length === 0)) { return undefined; }
+  if (typeof(obj) === 'object' && (Object.keys(obj as object).length === 0 || Object.values(obj as object).filter(x => x).length === 0)) { return undefined; }
```

Comment:
https://devblogs.microsoft.com/typescript/announcing-typescript-4-8/#unconstrained-generics-no-longer-assignable-to